From 21213af83d82fe9157234d36ec85e4dfc9c219db Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@strace.io>
Date: Wed, 10 Apr 2024 08:00:00 +0000
Subject: [PATCH 1/2] tests: prepare for --user=UID:GID syntax

* tests/options-syntax.test: Change the name of non-existent user
from ':nosuchuser:' to '!no such user!'.
---
 tests/options-syntax.test | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/tests/options-syntax.test b/tests/options-syntax.test
index cc82c6ce0..bc67377c7 100755
--- a/tests/options-syntax.test
+++ b/tests/options-syntax.test
@@ -345,23 +345,23 @@ diff -u -- "$EXP" "$LOG" > /dev/null || {
 uid="${UID:-`id -u`}"
 if [ "$uid" -ge 0 ]; then
 	if [ "$uid" -eq 0 ]; then
-		umsg="Cannot find user ':nosuchuser:'"
+		umsg="Cannot find user '!no such user!'"
 	else
 		if [ "$(fakeroot ../block_reset_run 15 id -u 2>&1)" = 0 ]; then
 			saved_STRACE=$STRACE
 			STRACE="fakeroot -- $STRACE"
-			umsg="Cannot find user ':nosuchuser:'"
-			check_e "$umsg" -u :nosuchuser: true
+			umsg="Cannot find user '!no such user!'"
+			check_e "$umsg" -u '!no such user!' true
 			STRACE=$saved_STRACE
 		fi
 		umsg='You must be root to use the -u/--username option'
 	fi
 
-	check_e "$umsg" -u :nosuchuser: true
+	check_e "$umsg" -u '!no such user!' true
 
 	for c in i/--instruction-pointer n/--syscall-number r/--relative-timestamps t/--absolute-timestamps T/--syscall-times y/--decode-fds; do
 		check_e "-$c has no effect with -c/--summary-only
-$STRACE_EXE: $umsg" -u :nosuchuser: -c -${c%%/*} true
+$STRACE_EXE: $umsg" -u '!no such user!' -c -${c%%/*} true
 	done
 
 	check_e "-i/--instruction-pointer has no effect with -c/--summary-only
@@ -371,47 +371,47 @@ $STRACE_EXE: -t/--absolute-timestamps has no effect with -c/--summary-only
 $STRACE_EXE: -T/--syscall-times has no effect with -c/--summary-only
 $STRACE_EXE: -y/--decode-fds has no effect with -c/--summary-only
 $STRACE_EXE: Only the last of -z/--successful-only/-Z/--failed-only options will take effect. See status qualifier for more complex filters.
-$STRACE_EXE: $umsg" -u :nosuchuser: -cinrtTyzZ true
+$STRACE_EXE: $umsg" -u '!no such user!' -cinrtTyzZ true
 
 	if [ -n "$compiled_with_secontext" ]; then
 		check_e "--secontext has no effect with -c/--summary-only
-$STRACE_EXE: $umsg" -u :nosuchuser: -c --secontext true
+$STRACE_EXE: $umsg" -u '!no such user!' -c --secontext true
 	else
 		check_e "SELinux context printing (--secontext option) is not supported by this build of strace" -c --secontext true
 	fi
 
 	if [ -n "$compiled_with_stacktrace" ]; then
 		check_e "-k/--stack-trace has no effect with -c/--summary-only
-$STRACE_EXE: $umsg" -u :nosuchuser: -c -k true
+$STRACE_EXE: $umsg" -u '!no such user!' -c -k true
 	fi
 
 	for c in --output-separately -A/--output-append-mode; do
 		check_e "$c has no effect without -o/--output
-$STRACE_EXE: $umsg" -u :nosuchuser: ${c%%/*} true
+$STRACE_EXE: $umsg" -u '!no such user!' ${c%%/*} true
 	done
 
 	check_e "-S/--summary-sort-by has no effect without (-c/--summary-only or -C/--summary)
-$STRACE_EXE: $umsg" -u :nosuchuser: --summary-sort-by errors true
+$STRACE_EXE: $umsg" -u '!no such user!' --summary-sort-by errors true
 
 	check_e "--output-separately has no effect without -o/--output
 $STRACE_EXE: -A/--output-append-mode has no effect without -o/--output
-$STRACE_EXE: $umsg" -u :nosuchuser: --output-separately --output-append-mode true
+$STRACE_EXE: $umsg" -u '!no such user!' --output-separately --output-append-mode true
 
-	check_e "$umsg" -u :nosuchuser: -ff true
-	check_e "$umsg" -u :nosuchuser: --output-separately --follow-forks true
+	check_e "$umsg" -u '!no such user!' -ff true
+	check_e "$umsg" -u '!no such user!' --output-separately --follow-forks true
 
 	check_e "Only the last of -z/--successful-only/-Z/--failed-only options will take effect. See status qualifier for more complex filters.
-$STRACE_EXE: $umsg" -u :nosuchuser: -z --successful-only true
+$STRACE_EXE: $umsg" -u '!no such user!' -z --successful-only true
 	check_e "Only the last of -z/--successful-only/-Z/--failed-only options will take effect. See status qualifier for more complex filters.
-$STRACE_EXE: $umsg" -u :nosuchuser: --successful-only -Z true
+$STRACE_EXE: $umsg" -u '!no such user!' --successful-only -Z true
 	check_e "Only the last of -z/--successful-only/-Z/--failed-only options will take effect. See status qualifier for more complex filters.
-$STRACE_EXE: $umsg" -u :nosuchuser: --successful-only --failed-only true
+$STRACE_EXE: $umsg" -u '!no such user!' --successful-only --failed-only true
 
 	check_e "-n/--syscall-number has no effect with -c/--summary-only
-$STRACE_EXE: $umsg" --user=:nosuchuser: --syscall-number --summary-only true
+$STRACE_EXE: $umsg" --user='!no such user!' --syscall-number --summary-only true
 
 	check_e "-i/--instruction-pointer has no effect with -c/--summary-only
-$STRACE_EXE: $umsg" --user=:nosuchuser: --instruction-pointer --summary-only true
+$STRACE_EXE: $umsg" --user='!no such user!' --instruction-pointer --summary-only true
 fi
 
 check_e_using_grep 'ptrace_setoptions = 0x[[:xdigit:]]+' -d /
@@ -431,7 +431,7 @@ if [ -z "$compiled_with_stacktrace" ]; then
 		--stack-trace-frame-limit=1
 else
 	check_e "--stack-trace-frame-limit has no effect without -k/--stack-trace
-$STRACE_EXE: $umsg" -u :nosuchuser: --stack-trace-frame-limit=1 true
+$STRACE_EXE: $umsg" -u '!no such user!' --stack-trace-frame-limit=1 true
 	if [ -n "$compiled_with_libunwind" ]; then
 		check_e "Stack traces with source line information (-kk/--stack-trace=source option) are not supported by this build of strace" -kk
 		check_e "Stack traces with source line information (-kk/--stack-trace=source option) are not supported by this build of strace" --stack-trace=source
-- 
2.43.0

